File: 14-longest-common-prefix/14-longest-common-prefix.js
/**
 * @param {string[]} strs
 * @return {string}
 */
var longestCommonPrefix = function(strs) {
    if (strs.length === 0) {
        return '';
    }
    if (strs.length === 1) {
        return strs[0];
    }
    strs.sort();
    let prefix = '';
    for (let i = 0; i < strs[0].length; i++) {
        if (strs[0][i] === strs[strs.length - 1][i]) {
            prefix += strs[0][i];
        } else {
            break;
        }
    }
    return prefix;
};

File: 14-longest-common-prefix/14-longest-common-prefix.rb
# @param {String[]} strs
# @return {String}
def longest_common_prefix(strs)
    return "" if strs.empty?
    return strs[0] if strs.length == 1
    strs.sort!
    prefix = ""
    (0...strs[0].length).each do |i|
        if strs[0][i] == strs[-1][i]
            prefix += strs[0][i]
        else
            break
        end
    end
    prefix
end